jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1949,4 +1962,43 @@ private static void failed(Callback callback, Throwable failure)
             throw t;
         }
     }
+
+    private static class ThreadPoolWrapper implements ThreadPool
+    {
+        private final Executor _executor;
+
+        private ThreadPoolWrapper(Executor executor)
+        {
+            _executor = executor;
+        }
+
+        @Override
+        public void execute(Runnable command)
+        {
+            _executor.execute(command);
+        }
+
+        @Override
+        public void join()
+        {
+        }
+
+        @Override
+        public int getThreads()
+        {
+            return 0;
```

Comment:
Probably should do:
```suggestion
            return _executor instanceof ThreadPool threadpool ? threadpool.getThreads() : 0;
```

and same for other methods

Reply:
Ditto.